### ISSUE TITLE
APIG passthroughBehavior property

### ISSUE THREAD

**Issue by HyperBrain** [serverless/serverless#1168]
There is a new functionality in APIG that allows you to define how an APIG method will behave if the incoming content-type does not match any defined `requestTemplate` content-type.
The behavior can be defined in the `passthroughBehavior` REST API property when creating or updating an APIG method resource.
Allowed values are: `WHEN_NO_MATCH`, `WHEN_NO_TEMPLATES` and `NEVER`.
Serverless should support setting this property as the default behavior is to pass the raw body to the attached lambda which is not perfect especially for API methods that should only accept one specific content type (e.g. `application/x-www-urlencoded`).
See here https://docs.aws.amazon.com/apigateway/api-reference/resource/integration/

**Comment by HyperBrain** [comment]
@flomotlik @pmuens Support for this also should be considered for V1

**Comment by liwp** [comment]
I'd suggest also setting the default to `NEVER`, because that will give, IMO, nicer behavior by default:

`NEVER`, AFAICT, means that the developer must explicitly define all content types that the lambda function can handle. Any other content types will result in a `415 - Unsupported Media Type` response from API Gateway and the lambda function is never called. This means that the lambda function can assume stricter input and doesn't have to handle crazy edge cases.

Also, when `passthroughBehaviour` is set to something other than `NEVER` and the request body is not JSON, API Gateway fails to parse the body and returns a nasty `400 - Bad Request` error with a body of:

```
{"message": "Could not parse request body into json: Unrecognized token \foo\': was expecting (\'true\', \'false\' or \'null\')\n at [Source: [B@63157e69; line: 1, column: 9]"}
```

**Comment by HyperBrain** [comment]
I also agree to have `NEVER` as default, but propose to stick to that beginning with Serverless V1.

I explicitly did not use that in my PR because for the currently distributed 0.5 version it probably would break existing deployments just by updating. With V1 already appearing at the horizon this may lead to some disturbance. Doing the semantic change with a new major version can, in my opinion, better be explained to the users as they have to adjust their projects anyway.

**Comment by pmuens** [comment]
Thanks for pointing to this!

We're switching from an API based deployment to CloudFormation. Unfortunately CloudFormation has not implemented this (https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-apitgateway-method-integration.html) so we need to postpone this until it's available through CF.

**Comment by jackrk** [comment]
CloudFormation should be releasing support soon. In terms of the default option, WHEN_NO_TEMPLATES may actually be better. If NEVER is set and there are no templates set on an Integration, any request would be rejected wth 415 which might be confusing to a new developer. At the same time it would also raise awareness for stricter content-type mapping... So I can see either side. 

Also, another key reason for releasing this feature was to allow API owners to trust the authenticity of injected data in their mapping templates (from $context or $stageVariables). If the integration passes data through when there is no template match, a client could potentially guess the outgoing transformation on template matches then they could spoof the injected data. For some developers this feature adds additional security to their API.

**Comment by flomotlik** [comment]
@HyperBrain is this still something you guys need? It sounds interesting, but could probably be achieved through custom resources in serverless.yml

**Comment by flomotlik** [comment]
And of course including docs

**Comment by HyperBrain** [comment]
@flomotlik Yes. It is a proper way to secure the APIs as invalid content can be blocked upfront.
I did not check yet, if CF already supports it, but if that's the case it would fit into the CF deployment perfectly. There should be only a way to specify the behavior in the service or function configuration.

**Comment by pmuens** [comment]
Seems like APIG now support `passthroughBehavior` in CloudFormation 🎉 

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-apitgateway-method-integration.html

**Comment by flomotlik** [comment]
ok cool @HyperBrain . Added the help-wanted-easy label as I'm not sure if we'll get to it soon, but it should be generally easy to add for someone

**Comment by pmuens** [comment]
This was implemented with #2000 🎉
